packages/calcite-components/src/utils/focusTrapComponent.ts

```diff
@@ -58,9 +71,10 @@ export function connectFocusTrap(component: FocusTrapComponent, options?: Connec
   }
 
   const focusTrapOptions: FocusTrapOptions = {
-    clickOutsideDeactivates: true,
-    escapeDeactivates: false,
+    clickOutsideDeactivates: component.clickOutsideDeactivates ?? true,
```

Comment:
We could maybe possibly make this something like: 

```suggestion
    clickOutsideDeactivates: component.clickOutsideDeactivates ?? clickOutsideDeactivates.call(null, this),
```